### PR TITLE
feat: add search bar

### PR DESCRIPTION
**Pull request by maxdesalle**
This PR adds a search bar to the website so users can quickly look up a specific topic through our documentation. This is achieved using [Algolia](https://www.algolia.com/), which is the [recommended way](https://docusaurus.io/docs/search#using-algolia-docsearch) to implement search on [Docusaurus](https://docusaurus.io/).

[Here](https://docs-fmuklvd77-sablier.vercel.app/) is how it looks like in production.

### CODE REVIEW

**Comment by PaulRBerg** [comment]
Good idea to add a search bar! My knee-jerk reaction was to request changes to this PR when I saw the API key committed in clear text, but it looks like it is safe to commit it, as per the Docusaurus docs.

I made a commit to make a few stylistic polishes (alphabetical ordering in `docusaurus.config.js` and double quotes instead of single quotes). Besides that, there's only the code comment left above to clarify, and then I'll merge.

File: src/css/custom.css [hunk 1]
```diff
@@ -36,3 +36,28 @@
...
+  --docsearch-text-color: var(--ifm-font-color-base);
+  --docsearch-muted-color: var(--ifm-color-secondary-darkest);
+  --docsearch-modal-background: var(--ifm-background-color);
+  --docsearch-searchbox-background: var(--ifm-background-color);
+  --docsearch-searchbox-focus-background: var(--ifm-color-black);
+  --docsearch-hit-color: var(--ifm-font-color-base);
+  --docsearch-hit-active-color: var(--ifm-color-white);
+  --docsearch-hit-background: var(--ifm-color-emphasis-100);
+  --docsearch-footer-background: var(--ifm-background-surface-color);
+}
```
Comment (PaulRBerg) [inline src/css/custom.css:63]: Are you sure we need these, @maxdesalle? Looks like it is reapplying the defaults that should be applied by Docusaurus nonetheless?

Comment (maxdesalle) [reply src/css/custom.css:63]: Yes, we do @paulrberg. The difference is subtle, but is present as you can see on the following two images. The first one is how it is with the added CSS settings, the second is without. The first image follows the colors of the rest of the website better IMO.
CleanShot 2022-07-11 at 13 10 59@2x
CleanShot 2022-07-11 at 13 11 14@2x

Comment (PaulRBerg) [reply src/css/custom.css:63]: Ok, yeah, it's a subtle difference, but noticeable.

**Pull request merged**